test/e2e/scale.go

```diff
@@ -124,26 +132,31 @@ func ScaleToWithin(t *testing.T, scale int, duration time.Duration, latencies La
 					return false, nil
```

Comment:
You should return `ctx.Err()` here as well. Maybe move the `if ctx.Err()` check above this `if` to catch that as well.